File: src/test/java/net/loxal/soa/restkit/endpoint/ResolveIpAddressServiceIT.java
/*
 * Copyright 2014 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package net.loxal.soa.restkit.endpoint;

import net.loxal.soa.restkit.model.whoami.Host;
import org.junit.Before;
import org.junit.Test;

import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import static org.junit.Assert.assertEquals;

public class ResolveIpAddressServiceIT extends AbstractEndpointTest {
    @Before
    public void setUp() throws Exception {
        resourcePath = ResolveIpAddressService.RESOURCE_PATH;
    }

    @Test
    public void assureHostAddress() throws Exception {
        String hostName = "loxal.net";
        String loxalNetIpAddress = "216.239.32.21";

        Response response = prepareGenericRequest(ResolveIpAddressService.RESOURCE_PATH)
                .queryParam(ResolveIpAddressService.HOST_NAME_PARAM, hostName).request().get();

        assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
        assertEquals(MediaType.APPLICATION_JSON_TYPE, response.getMediaType());

        Host host = response.readEntity(Host.class);
        assertEquals(loxalNetIpAddress, host.getAddress());
        assertEquals(hostName, host.getName());
    }
}

File: src/test/kotlin/net/loxal/soa/restkit/endpoint/ResolveIpAddressServiceIT.kt
/*
 * Copyright 2015 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package net.loxal.soa.restkit.endpoint

import org.junit.Before
import org.junit.Test
import net.loxal.soa.restkit.model.whoami.Host
import javax.ws.rs.core.Response
import javax.ws.rs.core.MediaType
import kotlin.test.assertEquals
import kotlin.test.assertNotNull
import kotlin.test.assertTrue

public class ResolveIpAddressServiceIT : AbstractEndpointTest() {
    Before
    public fun setUp() {
        AbstractEndpointTest.resourcePath = ResolveIpAddressService.RESOURCE_PATH
    }

    Test
    public fun assureHostAddress() {
        val hostName = "loxal.net"

        val response = AbstractEndpointTest.prepareGenericRequest(ResolveIpAddressService.RESOURCE_PATH).queryParam(ResolveIpAddressService.HOST_NAME_PARAM, hostName).request().get()

        assertEquals(Response.Status.OK.getStatusCode().toLong(), response.getStatus().toLong())
        assertEquals(MediaType.APPLICATION_JSON_TYPE, response.getMediaType())

        val host = response.readEntity<Host>(javaClass<Host>())
        assertNotNull(host.address)
        assertTrue(host.address.length() > 6 && host.address.length() < 20)
        assertEquals(hostName, host.name)
    }
}